lms/djangoapps/student_account/views.py

```diff
@@ -592,3 +596,45 @@ def account_settings_context(request):
         } for state in auth_states if state.provider.display_for_login or state.has_account]
 
     return context
+
+
+class RecoverPasswordView(ViewSet):
+
+    """
+    **Use Cases**
+            Resets a password of a user an email address
+        **Example Requests**:
+            POST /account/recover-password
+            {"email": "staff@example.com"}
+
+        **Response Values**
+            200 or 400
+    """
+
+    def post(self, request, format=None):
+        data = request.data
+        if 'email' in data:
+            limiter = BadRequestRateLimiter()
+            if limiter.is_rate_limit_exceeded(request):
+                AUDIT_LOG.warning("Password reset rate limit exceeded")
+                return HttpResponseForbidden()
+
+            email = data['email']
+
+            if len(email) > 0:
+
+                try:
+                    request_password_change(email, request.get_host(), request.is_secure())
+                    user = User.objects.get(email=email)
+                    destroy_oauth_tokens(user)
+                    return Response(status=status.HTTP_200_OK)
+                except UserNotFound:
+                    AUDIT_LOG.info("Invalid password reset attempt")
+                    # Increment the rate limit counter
+                    limiter.tick_bad_request_counter(request)
+                    return Response(status=status.HTTP_400_BAD_REQUEST)
+            else:
+                return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        else:
```

Comment:
I think if we drop the `else` block and leave the 400 return status below under no block, then we can remove the other `Response(status=status.HTTP_400_BAD_REQUEST)`s above and let them fall through to this last one by default because they didn't return a 200.